### DISCUSSION TITLE
Error while deployment: The runtime parameter of nodejs12.x is no longer supported for creating or updating AWS Lambda functions. The runtime used is nodejs18.x

### DISCUSSION THREAD

**Discussion by aryakrishn** [serverless/serverless#12053] [Q&A]
When I included cognito user pool event to one of the functions as follows,

- cognitoUserPool:
          pool: ${env:COGNITO_POOL_NAME}
          trigger: PostConfirmation
          existing: true
          forceDeploy: true

I am getting error on deployment. Without this event deployment is working.

The error is as follows,

An error occurred: CustomDashresourceDashexistingDashcupLambdaFunction - Resource handler returned message: "The runtime parameter of nodejs12.x is no longer supported for creating or updating AWS Lambda functions. We recommend you use the new runtime (nodejs18.x) while creating or updating functions.

However the runtime value in serverless.yml is   

runtime: nodejs18.x

What is the issue here?

**Comment by vibhoragarwal** [comment]
You must be using serverless <=2.72.2 (refer to serverless.yml and package.json)
The support was removed by AWS completely recently (May-June 2023 seemingly)

serverless provides a patch for this, but you need to upgrade to 2.72.4, and in case you can - upgrade to version 3 (3.330+)

See
https://github.com/serverless/serverless/releases

[2.72.4 (2023-07-10)](https://github.com/serverless/serverless/releases/tag/v2.72.4) Pre-release
Bug Fixes
custom resource runtime upgrade ([#12055](https://github.com/serverless/serverless/issues/12055)) ([fb1d29c](https://github.com/serverless/serverless/commit/fb1d29ca243b88bf3206e1b05f5a69eccae29052))

https://github.com/serverless/serverless/pull/12055


AWS deprecated the node js v12 lambda runtime. This PR upgrades to node 16

Closes: [#12054](https://github.com/serverless/serverless/issues/12054)

https://github.com/serverless/serverless/issues/12054

[@aryakrishn](https://github.com/aryakrishn) it is important to note this has been addressed in the most recent version of framework v3.330 but we have also applied the fix to v2.72.4 incase there is a reason you are unable to upgrade to v3.
